regression/cbmc-incr-smt2/pointer_arithmetic/pointer_subtraction_diff_types.c

```diff
@@ -0,0 +1,10 @@
+#include <stdlib.h>
+int main()
+{
+  int *x = malloc(sizeof(int));
+  float *y = x + 3;
+  int z = y - x;
```

Comment:
This subtraction is actually non-compiling code if I run it through `gcc`. If I run cbmc with the sat backend then this code is currently accepted and analysed. As this is not valid code it should really be detected in the C type checking performed as part of the C front end. Adding such front-end checks would be outside of the scope of this PR however.